consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/TransitionProtocolSchedule.java

```diff
@@ -181,37 +157,16 @@ private ProtocolSpec getByBlockHeaderFromTransitionUtils(
     return getPostMergeSchedule().getByBlockHeader(blockHeader);
   }
 
-  /**
-   * Gets by block number.
-   *
-   * @param number the number
-   * @return the by block number
-   */
-  @Override
-  public ProtocolSpec getByBlockNumber(final long number) {
-
-    return Optional.ofNullable(protocolContext)
-        .map(ProtocolContext::getBlockchain)
-        .flatMap(blockchain -> blockchain.getBlockHeader(number))
-        .map(timestampSchedule::getByBlockHeader)
-        .orElseGet(
-            () ->
-                transitionUtils.dispatchFunctionAccordingToMergeState(
-                    protocolSchedule -> protocolSchedule.getByBlockNumber(number)));
-  }
```

Comment:
getByTimestamp and getByBlockNumber are stitched together inside UnifiedProtocolSchedule.getByBlockHeader now, so complexity is encapsulated in that one place.